### PR TITLE
Metrics addon

### PR DESCRIPTION
**Pull request by libretto**
This addon enhances the statsd metrics available for monitoring the karapace server. We have now included the following additional metrics:

    connections_active
    request_size_avg
    request_size_max
    response_size_avg
    response_size_max

The aiohttp library does not provide native support for metrics. Therefore, we will make an effort to gather the required data by collecting statistics from the request handler, aiohttp logs, and various other sections of the karapace code.
We have plans to enhance the metrics support in this module by adding additional metrics.

### CODE REVIEW

File: requirements/requirements.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #
-# This file is autogenerated by pip-compile with Python 3.8
+# This file is autogenerated by pip-compile with Python 3.9
```
Comment (aiven-anton) [inline requirements/requirements.txt:2]: issue: Requirements need to be generated with Python 3.8, we should ideally have a CI check that validates this.

**Comment by tvainika** [comment]
Need to look in more detail, but just initial comments:
1. New features need some documentation, maybe a chapter in README to explain this feature
2. Some configuration to enable/disable this could be useful if not everyone wants this? Maybe could be related to fixing #593 ie. disabling statsd in full.
3. Can you explain rationale for calculating avg and max in karapace side instead of sending values as is, and letting metrics handled calculate e.g. average over given time period instead of something fixed here. Also in my quick look this implementation causes quite lot inaccuracies w.r.t. how time is handled. Sameple windows are resetted based on new data, instead per time frequency, or did I get this wrong?

**Comment by libretto** [comment]
@tvainika I am currently working on expanding the functionality of karapace metrics, which is already in existence. My objective is to incorporate metrics that are similar to what can be found in competitor products. To achieve this, I am utilizing the calculation methods employed by concurrent projects. These projects utilize the org.apache.kafka.common.metrics Java library, which includes static functions for calculating metrics such as Avg, Max, Min, and more. However, I haven't been able to find a Python library that offers similar functionality. Therefore, the metrics library I propose is a straightforward Python implementation based on the methods used in the aforementioned Java library.

While I have not personally explored the reasoning behind calculating average (avg) and minimum (min) metrics instead of sending them to a statsd server, it appears that both the Kafka team and competitors of karapace have found this approach to be rational. Despite the possibility of this method seeming irrational, it is crucial to provide a compatible solution for current customers who are migrating from competitors. Even if certain aspects may appear irrational, it is important to accommodate the preferences and expectations of customers to ensure a seamless transition to our platform.

About config: I will add config there.

**Comment by libretto** [comment]
Created the other branch #631

**Pull request closed without merging**